src/anyp/PortCfg.cc

```diff
@@ -102,12 +105,26 @@ AnyP::PortCfg::codeContextGist() const
 std::ostream &
 AnyP::PortCfg::detailCodeContext(std::ostream &os) const
 {
-    // parsePortSpecification() defaults optional port name to the required
-    // listening address so we cannot easily distinguish one from the other.
-    if (name)
-        os << Debug::Extra << "listening port: " << name;
-    else if (s.port())
-        os << Debug::Extra << "listening port address: " << s;
+    if (name_) {
+        os << Debug::Extra << "listening port: " << name_;
+    } else {
+        assert(spec_);
+        const auto scheme = AnyP::UriScheme(transport.protocol).image();
+        os << Debug::Extra << "listening port specification: " << scheme << "_port " << spec_;
+    }
+
     return os;
```

Comment:
I suggest removing the above and related name_/spec_ changes from this branch, leaving official port reporting code untouched and keeping just the "port iteration with context switching" parts. No need to "fix" the reporting lines in callers during this branch iteration -- that extra/double port reporting code will be removed when this branch is later "rebased". We can always restore the removed later if needed, but I think our TrafficMode printing work in #162 offers a better version of these reporting-related changes.